### PR TITLE
Feature/swa

### PR DESCRIPTION
**Pull request by MrZyr0**

### CODE REVIEW

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -3,6 +3,7 @@ on:
   push:
     branches:
       - main
+      - feature/swa
```
Comment (MaxenceMottard) [inline .github/workflows/deploy.yml:6]: remove

File: src/controllers/authentication.controller.ts [hunk 2]
```diff
@@ -37,4 +39,31 @@ export class AuthenticationController {
...
+      response_mode: 'form_post',
+      client_id: config.swa.serviceId,
+      redirect_uri: 'https://api.sign.quiches.ovh/auth/apple-id',
+      // state: '32ba49aa07', // TODO: need generation
+      scope: 'fullName email',
+    };
+
+    const searchParams = Object.keys(params)
+      .map((key) => {
+        return `${key}=${encodeURIComponent(params[key])}`;
```
Comment (MaxenceMottard) [inline src/controllers/authentication.controller.ts:63]: arrow function

File: src/modules/app.module.ts [hunk 3]
```diff
@@ -11,10 +11,12 @@ import { AWSService } from '../services/aws.service';
 import { FileController } from '../controllers/file.controller';
 import { GroupController } from '../controllers/group.controller';
 import { GroupService } from '../services/group.service';
+import { AppService } from '../services/app.service';
 
 @Module({
   imports: [MongooseModule.forRoot(config.mongoUrl), MongoModule],
   providers: [
+	AppService,
```
Comment (MaxenceMottard) [inline src/modules/app.module.ts:19]: indent

File: src/services/authentication.service.ts [hunk 4]
```diff
@@ -54,9 +57,29 @@ export class AuthenticationService {
...
+      lastname: parameters.lastname,
+    });
+
+    return { token: this.createToken(user) };
+  };
+
   private createToken = (user: User): string =>
-    jwt.sign({ _id: user._id }, config.jwt.secretKey, {
-      expiresIn: config.jwt.tokenExpirationTime,
+    jwt.sign({ _id: user._id, email: user.mail }, config.jwt.secretKey, {
```
Comment (MaxenceMottard) [inline src/services/authentication.service.ts:81]: remove email

**Review by MaxenceMottard — CHANGES_REQUESTED** [review]

File: .github/workflows/deploy.yml (hunk 1 above)
Comment (MrZyr0) [reply .github/workflows/deploy.yml:6]:
```suggestion
```

**Review by MaxenceMottard — APPROVED** [review]

**Pull request merged**